### ISSUE TITLE
May I ask if conv1d, conv2d, and batchnorm will be supported to implement the Clone trait?

### ISSUE THREAD

**Issue by kingzcheung** [huggingface/candle#909]
https://github.com/huggingface/candle/blob/f685b2231cad71e98257b62618b364c255feacd7/candle-nn/src/conv.rs#L24
https://github.com/huggingface/candle/blob/f685b2231cad71e98257b62618b364c255feacd7/candle-nn/src/conv.rs#L92
https://github.com/huggingface/candle/blob/f685b2231cad71e98257b62618b364c255feacd7/candle-nn/src/batch_norm.rs#L42

**Comment by EricLBuehler** [comment]
It could be easily implemented. You could wrap with Arc to achieve this, or construct a new ] Conv1d/2d/etc manually by constructing from the tensors.

**Comment by LaurentMazare** [comment]
It's certainly possible to wrap the layers in some `Arc` or `Rc` to make it cloneable, that said the `Tensor` type itself uses an `Arc` so it cloning of these layers should be fairly cheap so I've made #910 to implement this trait for the layer where it makes sense.

**Comment by kingzcheung** [comment]
> It's certainly possible to wrap the layers in some `Arc` or `Rc` to make it cloneable, that said the `Tensor` type itself uses an `Arc` so it cloning of these layers should be fairly cheap so I've made #910 to implement this trait for the layer where it makes sense.

This is an excellent job
